nats.go

```diff
@@ -857,6 +858,40 @@ func Secure(tls ...*tls.Config) Option {
 	}
 }
 
+// ClientTLSConfig is an Option to set the TLS configuration for secure
+// connections. It can be used to e.g. set TLS config with cert and root CAs
+// from memory. For simple use case of loading cert and CAs from file,
+// ClientCert and RootCAs options are more convenient.
+// If Secure is not already set this will set it as well.
+func ClientTLSConfig(certCB TLSCertHandler, rootCAsCB RootCAsHandler) Option {
+	return func(o *Options) error {
+		o.Secure = true
+
+		if certCB == nil && rootCAsCB == nil {
+			return ErrClientCertOrRootCAsRequired
+		}
+
+		// Smoke test the callbacks to fail early
```

Comment:
Its nice to smoke test it, but at the same time, it can be a problem in cases where the certs might be loaded a bit later (in a race'y environment)? Which might be a complexity in itself. Not saying no, just bringing @piotrpio attention :).

Reply:
I agree, but in such case this would be a problem even without the smoke test right? I mean, it makes sense to make sure to have the certs available before attempting to connect the client.

Reply:
Plus we do those smoke tests for other options (e.g. `UserCredentials`), so it makes sense for it to be consistent I think.